python/max_flow.py

```diff
@@ -0,0 +1,172 @@
+    """
+
+    records = get_paths(context, start_v, end_v, edge_property)
+
+    max_flow = 0
+    for record in records:
+        max_flow += record.fields["flow"]
+
+    return mgp.Record(max_flow=max_flow)
+
+
+@mgp.read_proc
+def get_paths(
+    context: mgp.ProcCtx,
+    start_v: mgp.Vertex,
+    end_v: mgp.Vertex,
+    edge_property: str = "weight",
+) -> mgp.Record(path=mgp.Path, flow=mgp.Number):
+    """
+    Returns each path and its flow used in max flow of a graph. Uses
+    Ford-Fulkerson algorithm, with capacity scaling for augmenting path
+    finding. Works for positive number weight values.
+
+    :param start_v: source vertex for outgoing flow
+    :param end_v: sink vertex for ingoing flow
+    :param edge_property: property of edge to be used as flow capacity
+
+    return: number value of graph max flow
+
+    The procedure can be invoked in openCypher using the following call:
+    MATCH (source {id: 0}), (sink {id: 5})
+    CALL max_flow.get_paths(source, sink, "weight")
+    YIELD path, flow
+    RETURN path, flow
+    """
+
+    if not isinstance(start_v, mgp.Vertex) or not isinstance(end_v, mgp.Vertex):
+        return [mgp.Record(path=None, flow=0)]
+
+    max_weight, min_weight = BFS_find_weight_min_max(start_v, edge_property)
+
+    if max_weight <= 0:
+        return [mgp.Record(path=None, flow=0)]
+
+    # delta is init as largest power of 2 smaller than max_weight
+    delta = 2 ** floor(log2(max_weight))
+
+    edge_flows = dict()
+    return_paths_and_flows = []
+
+    while True:
+        # augmenting path is a list of interchangeable
+        # VertexId and EdgeId
+        augmenting_path = [start_v.id]
+        flow_bottleneck = DFS_path_finding(
+            context, augmenting_path, start_v, end_v, edge_property, delta, edge_flows
+        )
+
+        if flow_bottleneck == -1:
+            if delta < min_weight:
+                break
+            delta //= 2
+            continue
+
+        for i, e in enumerate(augmenting_path):
+            if isinstance(e, mgp.Edge):
+                if augmenting_path[i - 1] == e.from_vertex.id:
+                    edge_flows[e.id] = edge_flows.get(e.id, 0) + flow_bottleneck
+                elif augmenting_path[i - 1] == e.to_vertex.id:
+                    edge_flows[e.id] = edge_flows.get(e.id, 0) - flow_bottleneck
+                else:
+                    raise Exception("path is not ordered correctly")
+
+        print(augmenting_path)
+
+        # store path as mgp.Path to return
+        for i, elem in enumerate(augmenting_path):
+            if i == 0:
+                path = mgp.Path(context.graph.get_vertex_by_id(augmenting_path[0]))
+            else:
+                if isinstance(elem, mgp.Edge):
+                    path.expand(elem)
+
+        return_paths_and_flows.append((path, flow_bottleneck))
+
+    return [mgp.Record(path=path, flow=flow) for path, flow in return_paths_and_flows]
+
+
+def DFS_path_finding(
+    context: mgp.ProcCtx,
```

Comment:
You don't need context here since it is a user-function, not a Memgraph read procedure.